js_modules/dagit/packages/core/src/app/useWebsocketAvailability.tsx

```diff
@@ -3,23 +3,37 @@ import * as React from 'react';
 import {WebSocketContext} from './WebSocketProvider';
 
 export const useWebsocketAvailability = () => {
-  const {websocketURI} = React.useContext(WebSocketContext);
+  const {websocketClient} = React.useContext(WebSocketContext);
   const [status, setStatus] = React.useState<'attempting-to-connect' | 'error' | 'success'>(
     'attempting-to-connect',
   );
 
   // Determine whether WebSockets are available at all. If not, fall back to a version
   // that uses a polling query.
   React.useEffect(() => {
-    const ws = new WebSocket(websocketURI, 'graphql-ws');
-    ws.addEventListener('error', () => setStatus('error'));
-    ws.addEventListener('open', () => {
+    if (!websocketClient) {
+      return;
+    }
+
+    let cleanup = () => {};
+
+    if (websocketClient.status === WebSocket.OPEN) {
       setStatus('success');
-      ws.close();
-    });
+    } else if (websocketClient.status === WebSocket.CLOSED) {
+      setStatus('error');
+    } else {
+      const errUnlisten = websocketClient.onError(() => setStatus('error'));
+      const connUnlisten = websocketClient.onConnected(() => setStatus('success'));
+      cleanup = () => {
+        errUnlisten();
+        connUnlisten();
+      };
+    }
 
-    return () => ws.close();
-  }, [websocketURI]);
+    return () => {
+      cleanup();
+    };
+  }, [websocketClient]);
```

Comment:
I don't think we really need to make this one use the Apollo `WebSocketClient`. Its only purpose is to check whether WebSockets are available at all, after which point we don't use it again. So a vanilla `WebSocket` is all we should need.